src/pages/StocktakesPage.js

```diff
@@ -54,95 +42,111 @@ export const StocktakesPage = ({ routeName, currentUser, dispatch: reduxDispatch
     isAscending: false,
     modalKey: '',
     hasSelection: false,
-    currentUser,
-    reduxDispatch,
     usingPrograms: getAllPrograms(Settings, UIDatabase).length > 0,
-  });
+  };
+};
+
+export const StocktakesPage = ({ routeName, currentUser, dispatch: reduxDispatch, navigation }) => {
+  const [state, dispatch, instantDebouncedDispatch] = usePageReducer(
+    routeName,
+    {},
+    stateInitialiser
+  );
 
   const {
     data,
     dataState,
     sortBy,
     isAscending,
-    columns,
+    searchTerm,
     modalKey,
     hasSelection,
     usingPrograms,
+    keyExtractor,
+    columns,
+    PageActions,
   } = state;
 
-  const { PROGRAM_STOCKTAKE } = MODAL_KEYS;
+  const callback = useCallback(() => dispatch(PageActions.refreshData()), []);
+  // Listen to sync changing stocktake data - refresh if there are any.
+  useSyncListener(callback, ['Stocktake']);
+  // Listen to navigation focusing this page - fresh if so.
+  useNavigationFocus(callback, navigation);
+
+  const onRowPress = useCallback(stocktake => reduxDispatch(gotoStocktakeEditPage(stocktake)), []);
+  const onFilterData = value => dispatch(PageActions.filterData(value));
```

Comment:
Should these functions be memoised, too?

Reply:
in an ideal world they all would be - but that kinda defeats the purpose. If we feel the performance gains are neccessarry we can add them (I think these pages mightv'e been better as class based components, so memoizing all of this wouldn't be neccessary)